python/ray/autoscaler/aws/config.py

```diff
@@ -115,6 +115,14 @@ def _configure_key_pair(config):
         assert "KeyName" in config["head_node"]
         assert "KeyName" in config["worker_nodes"]
         return config
+    elif "RAY_AWS_SSH_KEY" in os.environ:
+        key_path = os.environ["RAY_AWS_SSH_KEY"]
+        filename = os.path.basename(key_path)
+        key_name = os.path.splitext(filename)[0]
+        config["auth"]["ssh_private_key"] = key_path
+        config["head_node"]["KeyName"] = key_name
+        config["worker_nodes"]["KeyName"] = key_name
+        return config
```

Comment:
Are we storing private keys on Jenkins? Is that safe to do? Somebody could submit a PR that just `cat`s the private key to see what it is.

Reply:
The way Travis does this is by only storing an encrypted version of the private key, which Travis can decrypt but others can't decrypt.

Reply:
we can store creds through the jenkins UI.  i'll work w/richard and get this sorted monday.